hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataInputNavigableJsonAdapter.java

```diff
@@ -88,34 +90,78 @@ private boolean isQuote() throws IOException {
         return input.readByte() == '"';
     }
 
-    private static class UTF8Reader extends Reader {
+    static class UTF8Reader extends Reader {
 
-        private final DataInput input;
+        private final CharsetDecoder utf8Decoder;
+        private final ByteBuffer inputBuffer;
+        // required to support read() for multibyte characters
```

Comment:
Shouldn't this say 'surrogate pairs' instead of multibyte characters? It took me a while to understand what is meant by 'multibyte' here (it may be that I still don't understand it :-)